File: src/backend/ponto/src/model/ponto_relogio_model.py
from src import db


class PontoRelogioModel(db.Model):
    __tablename__ = 'ponto_relogio'

    id = db.Column(db.Integer, primary_key=True)
    localizacao = db.Column(db.String(50))
    marca = db.Column(db.String(30))
    fabricante = db.Column(db.String(30))
    numero_serie = db.Column(db.String(50))
    utilizacao = db.Column(db.String(1))


    def mapping(self, data):
        self.id = data.get('id')
        self.localizacao = data.get('localizacao')
        self.marca = data.get('marca')
        self.fabricante = data.get('fabricante')
        self.numero_serie = data.get('numeroSerie')
        self.utilizacao = data.get('utilizacao')

    def serialize(self):
        return {
            'id': self.id,
            'localizacao': self.localizacao,
            'marca': self.marca,
            'fabricante': self.fabricante,
            'numeroSerie': self.numero_serie,
            'utilizacao': self.utilizacao,
        }

File: frontend/ponto/lib/app/data/model/ponto_relogio_model.dart
import 'dart:convert';
import 'package:pluto_grid/pluto_grid.dart';

import 'package:ponto/app/data/domain/domain_imports.dart';

class PontoRelogioModel {
	int? id;
	String? localizacao;
	String? marca;
	String? fabricante;
	String? numeroSerie;
	String? utilizacao;

	PontoRelogioModel({
		this.id,
		this.localizacao,
		this.marca,
		this.fabricante,
		this.numeroSerie,
		this.utilizacao,
	});

	static List<String> dbColumns = <String>[
		'id',
		'localizacao',
		'marca',
		'fabricante',
		'numero_serie',
		'utilizacao',
	];
	
	static List<String> aliasColumns = <String>[
		'Id',
		'Localizacao',
		'Marca',
		'Fabricante',
		'Numero Serie',
		'Utilizacao',
	];

	PontoRelogioModel.fromJson(Map<String, dynamic> jsonData) {
		id = jsonData['id'];
		localizacao = jsonData['localizacao'];
		marca = jsonData['marca'];
		fabricante = jsonData['fabricante'];
		numeroSerie = jsonData['numeroSerie'];
		utilizacao = PontoRelogioDomain.getUtilizacao(jsonData['utilizacao']);
	}

	Map<String, dynamic> get toJson {
		Map<String, dynamic> jsonData = <String, dynamic>{};

		jsonData['id'] = id != 0 ? id : null;
		jsonData['localizacao'] = localizacao;
		jsonData['marca'] = marca;
		jsonData['fabricante'] = fabricante;
		jsonData['numeroSerie'] = numeroSerie;
		jsonData['utilizacao'] = PontoRelogioDomain.setUtilizacao(utilizacao);
	
		return jsonData;
	}
	
	String objectEncodeJson() {
		final jsonData = toJson;
		return json.encode(jsonData);
	}

	plutoRowToObject(PlutoRow plutoRow) {
		id = plutoRow.cells['id']?.value;
		localizacao = plutoRow.cells['localizacao']?.value;
		marca = plutoRow.cells['marca']?.value;
		fabricante = plutoRow.cells['fabricante']?.value;
		numeroSerie = plutoRow.cells['numeroSerie']?.value;
		utilizacao = plutoRow.cells['utilizacao']?.value != '' ? plutoRow.cells['utilizacao']?.value : 'Ponto';
	}	

	PontoRelogioModel clone() {
		return PontoRelogioModel(
			id: id,
			localizacao: localizacao,
			marca: marca,
			fabricante: fabricante,
			numeroSerie: numeroSerie,
			utilizacao: utilizacao,
		);			
	}

	
}